### PR TITLE
istioctl describe auto-mtls support

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/17833 by handling both the 1.3 and 1.4 style of Pilot /debug/authenticationz output.

### CODE REVIEW

File: istioctl/cmd/describe.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ import (
 	"istio.io/istio/pkg/config/protocol"
 	"istio.io/istio/pkg/config/schemas"
 	"istio.io/istio/pkg/kube/inject"
+
```
Comment (ayj) [inline istioctl/cmd/describe.go]: group with other istio.io imports

File: istioctl/cmd/describe.go [hunk 2]
```diff
@@ -623,21 +625,40 @@ func printAuthn(writer io.Writer, pod *v1.Pod, debug envoy_v2.AuthenticationDebu
...
 		"HTTP/mTLS":   "PERMISSIVE",
 		"TLS":         "SIMPLE",
 		"custom mTLS": "custom mTLS",
 		"UNKNOWN":     "Unknown",
 	}
-	tlsType, ok := mTLSType[debug.ServerProtocol]
+	tlsType, ok := mTLSType13[debug.ServerProtocol]
 	if !ok {
 		tlsType = debug.ServerProtocol
 	}
```
Comment (ayj) [inline istioctl/cmd/describe.go]: Move these lines to line 646 after the conflictStatus == "AUTO" check.

Comment (esnible) [reply istioctl/cmd/describe.go]: Fixed

File: istioctl/cmd/describe.go (hunk 1 above)
Comment (esnible) [reply istioctl/cmd/describe.go]: Fixed

**Pull request merged**